### ISSUE TITLE
Problem: jobs and tasks are not exposed via our APIs

### ISSUE THREAD

**Issue by jrwdunham** [artefactual/archivematica#1113]
We need to expose the Job and Tasks database tables as resources in our REST API. This will help to make data about the performance of Archivematica more transparent and more easily accessible.

The work in [PR 357 of the Storage Service](https://github.com/artefactual/archivematica-storage-service/pull/357) is relevant to this.

**Comment by ross-spencer** [comment]
Notes here on the potential to provide stdout/stderr in more helpful ways: https://github.com/archivematica/Issues/issues/428 

I have also written up a PICS analysis in an attempt to describe how this type of endpoint could help with systems testing, such as that done in AMAUAT: https://docs.google.com/document/d/1kTiSfq6X3vcj8VYywyyea0GXMCHbYa858uVL8ZBIA_o/edit?usp=sharing

**Comment by kenoir** [comment]
What about https://microservices.io/patterns/data/event-sourcing.html?

Having your services emit events when performing operations would allow you to both review them in testing and provide the opportunity to react to those events later?

**Comment by joel-simpson** [comment]
@kenoir that's an interesting idea and we've talked about moving to that sort of model. It's pretty easy to think about the data we are trying to extract as events and event streaming could potentially help solve scale problems. 
In this case I don't think we're really talking about changing the actual microservices, but simply putting a REST API in front of the existing database. That's a relatively small change. It does make me wonder if we think about this API as if it were serving up event information, would it be possible to maintain even if we moved to event-sourcing for microservices at a later date...
